toolchain/lower/testdata/function/generic/call_different_associated_const.carbon

```diff
@@ -0,0 +1,96 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/lower/testdata/function/generic/call_different_associated_const.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/lower/testdata/function/generic/call_different_associated_const.carbon
+
+// Cannot coalesce the lowering for G specifics, as they have associated contants of different types.
+// Check different functions are still emitted when trying to deduplicate emitted definitons.
+
+interface I {
+  let T:! type;
+  // TODO: An interface with multiple constraints is not yet implemented in mangling.
+  // fn F() -> T;
+}
+
+fn G(U:! I) {
+  var x: U.T;
+  // let x: U.T = U.F();
```

Comment:
Should this commented-out code be removed? It seems beside the point of this test, which is about detecting the difference between two specifics of the same generic that only have a different associated types. Perhaps this code would be in a different test doing the same thing for calling different associated functions?

Reply:
I think different tests for multiple associated constants in an interface and associated functions instead of associated types makes more sense.